muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/HelperReferenceWrapper.java

```diff
@@ -152,20 +152,27 @@ public String toString() {
   class Factory {
     private final TypePool classpathPool;
     private final Map<String, ClassRef> helperReferences;
+    private final HelperClassPredicate helperClassPredicate;
 
-    public Factory(TypePool classpathPool, Map<String, ClassRef> helperReferences) {
+    public Factory(
+        TypePool classpathPool,
+        Map<String, ClassRef> helperReferences,
+        HelperClassPredicate helperClassPredicate) {
       this.classpathPool = classpathPool;
       this.helperReferences = helperReferences;
+      this.helperClassPredicate = helperClassPredicate;
     }
 
     public HelperReferenceWrapper create(ClassRef reference) {
       return new ReferenceType(reference);
     }
 
     private HelperReferenceWrapper create(String className) {
-      Resolution resolution = classpathPool.describe(className);
-      if (resolution.isResolved()) {
-        return new ClasspathType(resolution.resolve());
+      if (!helperClassPredicate.isHelperClass(className)) {
+        Resolution resolution = classpathPool.describe(className);
```

Comment:
Looking up an injected helper records it in `AgentCachingPoolStrategy` as not found. If that helper is defined before it gets evicted there will be a stack trace about resource not found.

Reply:
this is helpful explanation, can you turn it into code comment? (especially since breaking it won't fail tests, only log annoying debug messages)